### PR TITLE
EXTREME MAKEOVER: Free the Tesla Gun edition

### PR DESCRIPTION
**Pull request by novaepee**
## About The Pull Request

Rebalanced, resprited, reworked, and RELEASED tesla gun!

Adding 2 tesla guns in armament vendor for marines to get instead of buying them from req.

## Why It's Good For The Game

The tesla gun is picking up spider webs and dust in the req order and development because (1) no one wants to buy them, (2) barely anyone knows it existence, meaning (3) barely, if at all no one, anyone knows how to use it in the current meta. 

I provide two free tesla guns to test out the water of how often marines grab it and go.

BALANCE STUFF:
- Giving it more ammo by having it shoot 4 time before reloading since its previous ammo count is 2 shots, which is pretty yikes
- Change from tesla gun having 20 seconds of disabling plasma regen to 10 seconds since 20 seconds is too much.

Power creep, but at least it doesn't cause damage to xenomorphs if the ball doesn't hit them; it just ruin their plan. It is also a good soft counter against primordial when they are unlock.

~~Tesla gun is currently under maintained, so expect jank from it. This PR is not to fix telsa gun but to make it more accessable.~~

I will update the tesla gun.

https://i.redd.it/rdlbpj6n68301.gif

## Changelog
Nova_epee :cl:
sprite: changin tesla sprite to spin
balance: tesla gun's ammo is now 4 instead of 2
balance: Tesla gun now disable plasma regen from 20 to 10 seconds.
expansion: add two tesla guns, a weapon that drain xenomorph's plasma, into armament  vendor under speciality tab; you can buy more at req
/:cl:

### CODE REVIEW

**Comment by neohenry** [comment]
yay, first pr

**Comment by BraveMole** [comment]
"Tesla gun is currently under maintained, so expect jank from it. This PR is not to fix telsa gun but to make it more accessable." 

Honestly you need to work on the testla gun first, it's jank

**Comment by hyper2snyper** [comment]
Define 'Jank'

**Comment by novaepee** [comment]
> Define 'Jank'

Check Discord tag.

**Comment by TiviPlus** [comment]
@Kuro020

**Comment by novaepee** [comment]
https://www.youtube.com/watch?v=z5kfoLcyyp4

Check video to see in its beauty.

**Comment by TiviPlus** [comment]
Checks failing

File: code/modules/projectiles/guns/rifles.dm [hunk 1]
```diff
@@ -1026,11 +1026,10 @@
 	)
 
 	flags_gun_features = GUN_AMMO_COUNTER|GUN_WIELDED_FIRING_ONLY|GUN_IFF
-	gun_firemode_list = list(GUN_FIREMODE_SEMIAUTO, GUN_FIREMODE_AUTOMATIC)
+	gun_firemode_list = list(GUN_FIREMODE_SEMIAUTO GUN_FIREMODE_AUTOMATIC, GUN_FIREMODE_AUTOBURST)
```
Comment (hyper2snyper) [inline code/modules/projectiles/guns/rifles.dm:1029]:
```suggestion
	gun_firemode_list = list(GUN_FIREMODE_SEMIAUTO, GUN_FIREMODE_AUTOMATIC, GUN_FIREMODE_AUTOBURST)
```

**Review by hyper2snyper — CHANGES_REQUESTED** [review]

File: code/modules/projectiles/guns/rifles.dm (hunk 1 above)
Comment (novaepee) [reply code/modules/projectiles/guns/rifles.dm]: Will give smartrifle burst fire later.

**Pull request merged**